### PR TITLE
Add Correlation ID to header for external tracking

### PR DESCRIPTION
**Pull request by wottop**
Added to logging for specific REST session
Stored on async jobs spawned by REST call

### CODE REVIEW

**Comment by barnabycourt** [comment]
I'm seeing some strangeness in the background/startup processes:
2017-02-28 16:21:38,812 [thread=localhost-startStop-1] [=, org=, csid=] INFO  org.candlepin.audit.HornetqContextListener - Hornetq server started

Is there a way to drop the  [=, org=, csid=]  entirely?

**Comment by wottop** [comment]
I don't know. I only added the last of those. The other name/values were already getting generated. I can look if you like.

File: server/src/main/java/org/candlepin/pinsetter/tasks/ImportJob.java [hunk 1]
```diff
@@ -124,6 +126,8 @@ public static JobDetail scheduleImport(Owner owner, String storedFileId, String
         map.put(STORED_FILE_ID, storedFileId);
         map.put(UPLOADED_FILE_NAME, uploadedFileName);
         map.put(CONFLICT_OVERRIDES, overrides.asStringArray());
+        map.put(JobStatus.CORRELATION_ID, MDC.get("csid"));
+        log.info("CSID: " + map.get(JobStatus.CORRELATION_ID));
```
Comment (barnabycourt) [inline server/src/main/java/org/candlepin/pinsetter/tasks/ImportJob.java]: Is this log statement necessary?

File: common/src/main/java/org/candlepin/common/filter/LoggingFilter.java [hunk 2]
```diff
@@ -72,6 +75,25 @@ public void doFilter(ServletRequest request, ServletResponse response,
...
+            while (headerNames.hasMoreElements()) {
+                String name = (String) headerNames.nextElement();
+                if ("X-Correlation-ID".equalsIgnoreCase(name)) {
+                    correlationId = ((HttpServletRequest) request).getHeader(name);
+                }
+            }
+
+            if (StringUtils.isAlphanumeric(correlationId) &&
+                correlationId.length() <= CORRELATION_ID_LENGTH) {
+                MDC.put("csid", correlationId);
```
Comment (barnabycourt) [inline common/src/main/java/org/candlepin/common/filter/LoggingFilter.java]: Please use a constant for "csid"

**Pull request merged**